doc/read-the-docs-site/howtos/asdata.rst

```diff
@@ -0,0 +1,89 @@
+.. highlight:: haskell
+.. _asdata:
+
+How to use ``AsData`` to optimize scripts
+=========================================
+
+Values stored in datums or redeemers need to be encoded into ``Data`` objects.
+This means that when a script starts to work with its datum or redeemer, it often wants to parse it into a more structured format using Plutus Tx's support for datatypes.
+Usually this is done with the ``fromBuiltinData`` function.
+
+The problem is that ``fromBuiltinData`` has to traverse the entire ``Data`` object in order to perform the transformation, which is costly.
+The alternative is to work directly with the raw ``Data`` object.
+
+These two approaches have tradeoffs:
+
+1. If the resulting object is going to be processed in its entirety, or have parts of it repeatedly processed, then it can be better to convert it, since Plutus Tx's datatypes are faster to work with than ``Data``.
+2. If it is important to check that the entire structure is well-formed, then it is better to convert it, since the conversion will check for well-formedness.
+3. If correctness is important, then it can be better to convert it, since the compiler can help you more if you are using Plutus Tx's datatypes.
+
```

Comment:
```suggestion
When conversion to Plutus Tx's datatypes is appropriate
-------------------------------------------------------

It is appropriate to convert an object to Plutus Tx's datatypes under the following circumstances: 

1. When the resulting object is going to be processed in its entirety, or have parts of it repeatedly processed, since Plutus Tx's datatypes are faster to work with than ``Data``.
2. If it is important to check that the entire structure is well-formed, since the conversion will check for well-formedness.
3. If correctness is important, since the compiler can help you more if you are using Plutus Tx's datatypes.

```